### PR TITLE
Convert Tests To Jest

### PR DESCRIPTION
**Pull request by Zyie**
This PR converts all the tests to Jest

It also adds a couple of new commands and removes old ones

- `test` - runs jest normally
- `test:debug` - runs jest in electron window
- `test:pkg` - runs jest on specific packages e.g. `npm run test:pkg -- --packages=@pixi/math @pixi/core`
- `test:pkg:debug` - runs jest on specific packages but in electron window

I'm also piggy backing off of the Extensions API #8410 as it dealt with the global pollution of tests much better than my attempt

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
     "clean:modules": "lerna clean --yes",
     "clean": "run-s clean:*",
-    "floss": "cross-env TS_NODE_PROJECT=tsconfig.tests.json floss -r ts-node/register/transpile-only -p test",
-    "test": "run-s floss",
-    "unit-test": "npm run floss -- --",
-    "unit-test:debug": "npm run floss -- --debug --",
+    "test:pkg": "ts-node --transpile-only ./test/index.ts",
+    "test:pkg:debug": "cross-env DEBUG_MODE=1 ts-node --transpile-only ./test/index.ts",
+    "test": "npx jest --silent",
+    "test:debug": "cross-env DEBUG_MODE=1 npx jest",
```
Comment (GoodBoyDigital) [inline package.json:18]: should we stick a `--watch` in here too?

**Review by GoodBoyDigital — APPROVED** [review]
nice!

**Comment by bigtimebuddy** [comment]
Couple of things:

Coverage with nyc doesn't seem to be reporting anything, could you look into it? This has been a useful metric in the past to figure out where to focus unit-testing improvements.

```
--------------------------------------|---------|----------|---------|---------|-------------------
File                                  | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
--------------------------------------|---------|----------|---------|---------|-------------------
All files                             |       0 |        0 |       0 |       0 |                   
```

Also, the test time is about 33% longer (~1:30 vs ~2:00). Is there any opportunity to speed-up the test run?

**Comment by bigtimebuddy** [comment]
Could you resolve conflicts @Zyie?

**Comment by bigtimebuddy** [comment]
Looks like the test time increased from ~2 minutes to about ~4 minutes. Is that because coverage is now working or because of your speed-ups?

**Review by bigtimebuddy — APPROVED** [review]

**Comment by bigtimebuddy** [comment]
RIP to Floss, you've served us well bud 🍻

**Pull request merged**